$OpenBSD: patch-src_libgmtk_gmtk_media_player_h,v 1.3 2011/10/17 13:50:34 dcoppa Exp $

Raise event on incomplete QuickTime file, and mark it for retry
(fix apple.com/trailers) - upstream svn revision r2152

Fix speed multipliers and speed reset (upstream svn revisions r54
and r55 of gmtk)

--- src/libgmtk/gmtk_media_player.h.orig	Tue Jun 28 15:29:18 2011
+++ src/libgmtk/gmtk_media_player.h	Mon Oct 17 14:43:23 2011
@@ -170,7 +170,9 @@ typedef enum {
     ATTRIBUTE_ENABLE_FRAME_DROP,
     ATTRIBUTE_TITLE,
     ATTRIBUTE_ARTIST,
-    ATTRIBUTE_ALBUM
+    ATTRIBUTE_ALBUM,
+    ATTRIBUTE_RETRY_ON_FULL_CACHE,
+    ATTRIBUTE_SPEED_SET
 } GmtkMediaPlayerMediaAttributes;
 
 typedef enum {
@@ -201,6 +203,13 @@ typedef enum {
     COMMAND_SWITCH_FRAME_DROP
 } GmtkMediaPlayerCommand;
 
+typedef enum {
+    EVENT_TYPE_INT,
+    EVENT_TYPE_DOUBLE,
+    EVENT_TYPE_BOOLEAN,
+    EVENT_TYPE_ALLOCATION
+} GmtkMediaPlayerEventType;
+
 typedef struct _GmtkMediaPlayer GmtkMediaPlayer;
 typedef struct _GmtkMediaPlayerClass GmtkMediaPlayerClass;
 
@@ -222,9 +231,11 @@ typedef struct _GmtkMediaPlayerAudioTrack {
 
 typedef struct _GmtkMediaPlayerEvent {
     GmtkMediaPlayer *player;
+    GmtkMediaPlayerEventType type;
     gchar *event_name;
     gint event_data_int;
     gdouble event_data_double;
+    gboolean event_data_boolean;
     GtkAllocation *event_allocation;
 } GmtkMediaPlayerEvent;
 
@@ -304,6 +315,7 @@ struct _GmtkMediaPlayer {
     gchar *album;
     gboolean enable_divx;
     gboolean disable_xvmc;
+    gboolean retry_on_full_cache;
 
     gboolean deinterlace;
     gboolean frame_drop;
